plugins/cloudservices/plugin.js

```diff
@@ -0,0 +1,127 @@
+			 * skipped {@link CKEDITOR.config#cloudServices_token} will be used.
+			 */
+			function CloudServicesLoader( editor, fileOrData, fileName, token ) {
+				FileLoader.call( this, editor, fileOrData, fileName );
+
+				/**
+				 * Custom [Cloud Service](https://ckeditor.com/ckeditor-cloud-services/) token.
+				 *
+				 * @property {String} customToken
+				 * @member CKEDITOR.plugins.cloudservices.cloudServicesLoader
+				 */
+				this.customToken = token;
+			}
+
+			CloudServicesLoader.prototype = CKEDITOR.tools.extend( {}, FileLoader.prototype );
+
+			/**
+			 * @inheritdoc
+			 * @param {String} [url] The upload URL. If not provided {@link CKEDITOR.config#cloudServices_url} will be used.
+			 * @param {Object} [additionalRequestParameters] Additional data that would be passed to the
+			 * {@link CKEDITOR.editor#fileUploadRequest} event.
+			 */
+			CloudServicesLoader.prototype.upload = function( url, additionalRequestParameters ) {
+				url = url || this.editor.config.cloudServices_url;
+
+				FileLoader.prototype.upload.call( this, url, additionalRequestParameters );
+			};
+
+			/**
+			 * @method loadAndUpload
+			 * @inheritdoc
+			 * @param {String} [url] The upload URL. If not provided {@link CKEDITOR.config#cloudServices_url} will be used.
+			 * @param {Object} [additionalRequestParameters] Additional parameters that would be passed to
+			 * the {@link CKEDITOR.editor#fileUploadRequest} event.
+			*/
+
+			CKEDITOR.plugins.cloudservices.cloudServicesLoader = CloudServicesLoader;
+		},
+
+		beforeInit: function( editor ) {
+			editor.on( 'fileUploadRequest', function( evt ) {
+				var fileLoader = evt.data.fileLoader,
+					reqData = evt.data.requestData;
+
+				if ( fileLoader instanceof CKEDITOR.plugins.cloudservices.cloudServicesLoader ) {
+					// Cloud Services expect file to be put as a "file" property.
+					reqData.file = reqData.upload;
+					delete reqData.upload;
+
+					// Add authorization token.
+					evt.data.fileLoader.xhr.setRequestHeader( 'Authorization', fileLoader.customToken || editor.config.cloudServices_token );
+				}
+			}, null, null, 6 );
+
+			editor.on( 'fileUploadResponse', function( evt ) {
+				var fileLoader = evt.data.fileLoader,
+					xhr = fileLoader.xhr,
+					response;
+
+				if ( fileLoader instanceof CKEDITOR.plugins.cloudservices.cloudServicesLoader ) {
+					evt.stop();
+
+					try {
+						response = JSON.parse( xhr.responseText );
+
+						evt.data.response = response;
+					} catch ( e ) {
+						CKEDITOR.warn( 'filetools-response-error', { responseText: xhr.responseText } );
+					}
+				}
+			} );
+		}
+	} );
+
+	CKEDITOR.plugins.cloudservices = {
+		// Note this type is loaded on runtime.
+		cloudServicesLoader: null
+	};
+
+	/**
+	 * Endpoint URL for [Cloud Services](https://ckeditor.com/ckeditor-cloud-services) uploads.
+	 *
+	 * @since 4.8.0
+	 * @cfg {String} [cloudServices_url='']
```

Comment:
This cfg is no longer valid.

Reply:
Why is no longer valid?

Reply:
Oh, I'm sorry I have referenced config variables in a wrong file, there was `easyimageUploadUrl` and `easyimage_token` in `easyimage/plugin.js` which I have removed in https://github.com/ckeditor/ckeditor-dev/pull/1052/commits/0d5dc0111d0b0a18eef0eaeb528b497bae2f763c